src/main/kotlin/io/github/gradlenexus/publishplugin/internal/Gradle5ArtifactRepositoryExtensions.kt

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright 2021 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.github.gradlenexus.publishplugin.internal
+
+import org.gradle.api.artifacts.repositories.ArtifactRepository
+import org.gradle.api.artifacts.repositories.IvyArtifactRepository
+import org.gradle.api.artifacts.repositories.MavenArtifactRepository
+import org.gradle.api.provider.Provider
+import java.net.URI
+
+/**
+ * Contains extension methods that mimic Gradle 6+ API for ArtifactRepository
```

Comment:
Is that needed only for Gradle 5 compatibility? If yes, I think, it should be easier to maintain to just make Gradle 6 (released on Nov 8, 2019) the minimal required version. WDYT?

Reply:
Yes this, and droping convention is required for gradle 5.0 compatibility. Raising to 6 should solve the problems

Reply:
Great, I'm for it.

Reply:
I think this would indeed be a good reason to drop compatibility with Gradle 5.x but in that case I think we should bump the major version number of the plugin to `2.0.0` to indicate a breaking change.

@maczikasz Could you please take care of that?